### PR TITLE
Refactor: plans and pricing promotional content

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
Resolves: https://app.asana.com/0/1205251546311816/1205651762721233

## Description

This PR allows promotional pricing content to be inserted into the Add-ons page during the BF event week notifying users of discounted sale's prices on plans. 

Sale's period:
```
'startDate' => '2023-11-20 00:00',
'endDate' => '2023-11-29 23:59',
```

## Affects
- Add-ons page

### Before:
Screen Shot 2023-11-06 at 10 11 57 AM

### After:
Screen Shot 2023-11-06 at 10 11 41 AM

## Pre-review Checklist

-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Promotions/InPluginUpsells/AddonsAdminPage.php [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 namespace Give\Promotions\InPluginUpsells;
 
+use DateTimeImmutable;
+use DateTimeZone;
+use Exception;
+use Give\Framework\Shims\Shim;
```
Comment (JasonTheAdams) [inline src/Promotions/InPluginUpsells/AddonsAdminPage.php:8]: @JoshuaHungDinh Is this being used? I can't even find what this references in the codebase.

Comment (JoshuaHungDinh) [reply src/Promotions/InPluginUpsells/AddonsAdminPage.php]: @JasonTheAdams No, it seems I inadvertently copied that in from the `salesbanner ` class. It was originally introduced in 8c591e48d09d7c7ecad5f97acb9875f5a3577f3e and later removed in https://github.com/impress-org/givewp/pull/6384/commits/98caab5976b2447ca01f9bd244e3da4f5a0ce15f. 

I've removed all of it's remaining imports Resolved: 8c591e48d09d7c7ecad5f97acb9875f5a3577f3e + 4f5aa753325997d7bd58c514318864eabbd442e8

**Review by JasonTheAdams — APPROVED** [review]
Looks good to me, @JoshuaHungDinh. Interesting that we don't really have a way of supporting this sort of thing in the addons.json file on givewp.com. But this is probably the way to go for now.

**Comment by JasonTheAdams** [comment]
@JoshuaHungDinh Is this in QA?

**Comment by JoshuaHungDinh** [comment]
> @JoshuaHungDinh Is this in QA?

@JasonTheAdams, I had moved it forward on the new board, but will ping QA. I noticed new subtasks included in here. Are we going to update the addons.json instead? https://stellarwp.atlassian.net/jira/software/c/projects/GIVE/boards/382?selectedIssue=GIVE-39

**Comment by JasonTheAdams** [comment]
Hey @JoshuaHungDinh! I'm actually going to close this PR, so we can pull it out of QA.

The reason is that we're deviating from how simple the add-ons content system is. The point is that it _doesn't_ require plugin updates in order to propagate content changes to plugins, and can just update the `addons.json` file on the givewp.com server. This change compensates for the fact that the system doesn't have a scheduler in it, but we can easily make the changes on the 20th, then revert them on the 29th, and it will automatically propagate to every GiveWP customer without requiring a plugin update. So we'll just do that. 😄

**Pull request closed without merging**